### PR TITLE
fix: trip endpoint spec gaps — routeShortName source and includeReferences

### LINKED ISSUE #987: Spec review: trip
## What to review
Read the [wiki spec](https://github.com/OneBusAway/maglev/wiki/trip) and compare the full Maglev implementation of this endpoint against it. Look for:

- **Request parameters**: all params supported with correct types, defaults, and validation
- **Response structure**: all fields present with correct names, types, and nesting
- **Error handling**: error cases match the spec (404, 400, etc.)
- **Behaviour**: the endpoint behaves as the spec describes, including edge cases
- **Test coverage**: main scenarios and edge cases from the spec are covered by tests

## Workflow
1. Compare the implementation against the wiki spec. If no gaps found, move this card to **Done**.
2. For each gap or bug, file a GitHub issue labelled `spec-gap`, link it to this card, and assign it to yourself.
3. Move this card to **Issues Filed**.
4. Work on the issues you filed.
5. Once all linked issues are closed, move this card back to **In Review** and repeat from step 1.

## Deeper analysis
The skills at https://github.com/burma-shave/maglev-workspace can help when the spec or implementation behaviour is not clear. They analyse the Java OBA reference implementation by downloading, building, and running the server — enabling both static and runtime analysis.

Install them into your coding agent (e.g. Claude Code). Key capabilities:
- **Q&A mode**: ask arbitrary questions about this endpoint and get answers grounded in the Java source and live server behaviour
- **Spec generation**: regenerate or cross-check the spec directly from the Java source

**Requirements**: Docker and Git
**Note**: the skills are token-intensive and trigger many permission prompts — a VM or sandbox environment is recommended

### PR DESCRIPTION
**Pull request by Ahmedhossamdev**
Fixes: #987, #1062

Fixes two spec gaps in `/api/where/trip/{id}`, both verified byte-for-byte against the Java reference server.

**1. `entry.routeShortName` sourced from the wrong table**
It read the route's short name; the spec/Java define it as the trip's own per-trip route short name (GTFS `trips.txt route_short_name`). Maglev doesn't store that column, so the parity-correct value is `""`. Clients fall back to `references.routes[].shortName` as the spec instructs.

**2. `includeReferences=false` ignored**
The handler never read the parameter and always populated references. Now parses it (default `true`) and returns empty reference arrays when `false`.

## Summary by CodeRabbit

* **New Features**
  * Added `includeReferences` query parameter to the trip endpoint. When set to `false`, reference data (routes, agencies, stops, trips, situations) are excluded from responses. References are included by default.

* **Bug Fixes**
  * Updated route short name field to reflect correct stored values.

* **Tests**
  * Added test coverage for the new `includeReferences` parameter functionality.

### CODE REVIEW

File: internal/restapi/trip_handler.go [hunk 1]
```diff
@@ -67,29 +72,40 @@ func (api *RestAPI) tripHandler(w http.ResponseWriter, r *http.Request) {
...
-		route.LongName.String,
-		route.Desc.String,
-		models.RouteType(route.Type),
-		route.Url.String,
-		route.Color.String,
-		route.TextColor.String))
+	// includeReferences defaults to true; when explicitly false the references
+	// block is returned with all sub-arrays empty (matches the Java reference).
+	includeReferences := true
+	if v := r.URL.Query().Get("includeReferences"); v != "" {
```
Comment (burma-shave) [inline internal/restapi/trip_handler.go]: There is a ShouldIncludeReferences() in reference_utils.go. This is a good place to use it.

**Review by burma-shave — CHANGES_REQUESTED** [review]

**Review by burma-shave — APPROVED** [review]
